Feat: 검사 결과 가치관 설명 카드 슬라이드 애니메이션 구현

sejinbaek opened the pull request:
## 📝 변경 사항

> 커밋 단위로 명시적으로 짧게 작성합니다.

1. 차트 라벨 클릭 시 가치관 설명 카드 등장 애니메이션을 추가하였습니다.
2. 카드 닫기 버튼 클릭 시 원위치로 돌아가는 애니메이션을 구현하였습니다.
3. AnimatePresence를 삭제하여 레이아웃 구조를 개선하였습니다. (동시 이동 타이밍 맞춤)
4. 라벨 클릭 시 active를 적용하여 라벨을 강조하였습니다.

## 🔍 변경 사항 세부 설명

> 세부 설명을 작성합니다.

- 차트 라벨 클릭 시 검사 결과와 연결된 가치관 설명 카드가 자연스럽게 등장하는 애니메이션을 추가했습니다.
- 카드 닫기(X) 버튼 클릭 시 차트가 원위치로 돌아가고, 카드가 사라지는 애니메이션을 구현했습니다.
- 카드의 등장/퇴장은 opacity 및 transform 기반 애니메이션으로 처리했습니다.
- 기존 AnimatePresence를 제거하고, 레이아웃(layout transition) 기반으로 차트와 카드가 동시에 움직이도록 구조를 리팩토링했습니다.
- 카드 영역은 항상 유지하면서, AnimatePresence로 인해 발생하던 exit 타이밍 지연 문제를 해결했습니다.
- 라벨 클릭 시 active된 상태일 경우, 폰트 크기 및 폰트 색상, 폰트 두께를 강조하였습니다.

## 🕵️‍♀️ 요청사항

> 팀원들에게 테스트나 리뷰를 요청합니다.

- 차트 라벨 클릭 시 카드가 자연스럽게 등장하는지 확인 부탁드립니다.
- 닫기 버튼 클릭 시 차트와 카드가 동시에 부드럽게 원위치로 이동하는지 확인 부탁드립니다.
- 애니메이션이 불필요하게 딜레이되거나 어색한 지점이 없는지 테스트 부탁드립니다.

## 📷 스크린샷 (선택)

> UI 변경이 있는 경우 스크린샷이나 GIF를 첨부합니다.
![Animation](https://github.com/user-attachments/assets/3805d486-9667-46e3-bafc-4770345b4aaa)


## ✅ 작성자 체크리스트

- [x] Self-review: commit 이나 오타 없이 코드가 스스로 검토됨
- [x] 로컬에서 모든 기능이 정상 작동함(버그수정 /기능에 대한 테스트)
- [x] 린터 및 포맷터로 코드 정리됨

sejinbaek committed:
Style: ResultDescriptionCard 레이아웃 및 섹션 타이틀 스타일링 추가

- Card 컴포넌트 반응형 width (clamp) 설정
- 내부 패딩, border-radius, box-shadow 적용
- SectionTitle 스타일링 (배경색, 보더, 그림자, 폰트 사이즈)

sejinbaek committed:
Style: ResultPage 윗부분 레이아웃 스타일 적용

sejinbaek committed:
Style: 가치관 특징 카드 배치를 위한 스타일 수정

sejinbaek committed:
Chore: 불필요한 코드 제거

sejinbaek committed:
Feat: ResultPage 컴포넌트 생성 및 애니메이션 적용

- Chart는 layout 애니메이션 적용으로 위치 자연스럽게 이동
- DescriptionCard는 AnimatePresence 및 slideInVariants로 슬라이드 인/아웃 구현
- spring 기반의 자연스러운 애니메이션 transition 설정

sejinbaek committed:
Feat: close 버튼 클릭 시 원위치로 이동하는 기능 구현

sejinbaek committed:
Feat: 카드 닫기 버튼 클릭 시 원위치로 이동하는 기능 구현

sejinbaek committed:
Chore: 닫기 버튼 아이콘 추가

sejinbaek committed:
Feat: 닫기 버튼 배치

sejinbaek committed:
Refactor: 카드 퇴장 시 차트와 동시 이동을 위해 AnimatePresence 제거 및 레이아웃 구조 개선

- 카드 퇴장 시 차트가 늦게 움직이는 문제를 해결하기 위해 AnimatePresence 제거
- 카드 영역을 항상 유지하는 방식으로 layout transition을 적용하여 차트와 카드가 동시에 자연스럽게 이동하도록 개선
- 카드가 사라질 때 공간은 유지하고, opacity와 transform을 이용해 등장/퇴장 효과를 처리
- ResultDescriptionCard를 감싸는 motion.div는 항상 존재하고, 카드만 조건부 렌더링되도록 변경
- 차트와 카드가 한 덩어리 layout으로 동작하여 exit 타이밍 이슈 해결

sejinbaek committed:
Fix: lint 에러 및 경고 수정 (props 타입 정의, alt 속성, 불필요 import 제거)

- ResultChart 컴포넌트 onLabelClick props 타입 정의 추가
- ResultDescriptionCard label 미사용 경고 해결 (미사용 시 제거)
- 닫기 버튼 이미지 alt 속성 추가 (접근성 대응)
- ResultPage에서 사용되지 않는 AnimatePresence import 삭제

sejinbaek committed:
Fix: ResultPage selectedLabel 타입 오류 및 ResultDescriptionCard props 타입 정의 추가

- ResultPage selectedLabel 타입을 string | null 로 변경하여 라벨 클릭 시 타입 오류 수정
- ResultDescriptionCard 컴포넌트에 label props 타입 정의 추가
- 타입 오류로 인한 빌드 실패 해결

sejinbaek committed:
Fix: 사용하지 않는 props 제거

sejinbaek committed:
Fix: ResultDescriptionCard props 타입 오류 및 setSelectedLabel 타입 오류 수정

- ResultDescriptionCard props 타입에 label: string 정의 추가
- ResultPage에서 setSelectedLabel(false) 를 setSelectedLabel(null)로 수정하여 타입 오류 해결
- 빌드 실패 원인 제거

sejinbaek committed:
Feat: 직업 가치관 설명 데이터(resultValuesData) 추가

sejinbaek committed:
Feat: 직업 가치관 설명 카드에 데이터 바인딩

hyonun321 commented on client/src/features/result/ResultDescriptionCard.tsx:
```diff
@@ -45,22 +46,26 @@ type ResultDescriptionCardProps = {
+  const result = resultValuesData.find((item) => item.type === label);
+  if (!result) return null;
+
+  const { value, choice, work } = result.description;
```
👍

hyonun321 reviewed (approved):
확인했습니다. 불필요한 코드 제거의 커밋인경우, 어떤 사유로 제거하게 됐는지 간단하게 작성해주셔도 좋을 것 같습니다. 수고하셨어요!

yeom-kenco commented:
선택된 단어에 색깔 변경 및 폰트 두께와 사이즈 조정으로 현재 모달이 어떤 단어를 클릭해서 나오는건지 표현되는 효과가 필요할 것 같습니다! 피그마에 디자인한 것 처럼요!

sejinbaek committed:
Style: 카드 오른쪽 패딩 추가

sejinbaek commented:
> 선택된 단어에 색깔 변경 및 폰트 두께와 사이즈 조정으로 현재 모달이 어떤 단어를 클릭해서 나오는건지 표현되는 효과가 필요할 것 같습니다! 피그마에 디자인한 것 처럼요!

넵 피드백 감사합니다 ! 수정해보겠습니다 ~!

> 확인했습니다. 불필요한 코드 제거의 커밋인경우, 어떤 사유로 제거하게 됐는지 간단하게 작성해주셔도 좋을 것 같습니다. 수고하셨어요!

넵 ! 자세히 커밋 메시지 작성하겠습니다 !

sejinbaek committed:
Style: 활성화된 라벨 색상 및 폰트 크기 스타일 추가

- 활성화된 라벨(activeLabel)에 대해 색상(primary) 및 폰트 크기 강조 스타일을 추가했습니다.
- 선택된 라벨은 기존 회색 → primary 색상으로 변경되고, 글씨 크기가 커집니다.
- hover와 구분되는 선택(active) 상태 표현을 명확히 개선했습니다.

sejinbaek committed:
Style: active 라벨 스타일을 theme 값으로 통일

- 라벨의 font-size, color, font-weight 스타일을 theme.js에서 정의한 값으로 일관성 있게 적용했습니다.
- 하드코딩된 색상 코드와 숫자값을 제거하고, 디자인 시스템 기준 (theme.fontSize, theme.colors, theme.fontWeight) 를 활용하도록 리팩토링했습니다.
- 활성화(active) 상태에서도 theme 값을 기준으로 스타일이 적용됩니다.

sejinbaek commented:
![Animation](https://github.com/user-attachments/assets/0f92b4f9-7f06-4aa5-ae20-1f7633b16127)
개선된 시연 영상입니다 !

yeom-kenco reviewed (approved):
구현하기 엄청 어려웠을거같은데ㅜㅜ대단합니다!!! 수정하신부분 확인했습니다. 최고!

H-JuKyung reviewed (approved):
수고 많으셨어요ㅜㅜ!!!👍🏻

hyonun321 committed:
Merge branch 'main' into Feat-검사결과-모달-구현

hyonun321 commented:
수고하셨습니다~!
